data-tck/src/main/java/io/micronaut/data/tck/entities/SaleItem.java

```diff
@@ -30,6 +31,7 @@ public class SaleItem {
 
     @Relation(value = Relation.Kind.MANY_TO_ONE, cascade = Relation.Cascade.ALL)
     @MappedProperty("fk_sale_id")
+    @JsonIgnore // To avoid infinite recursion when reading Sale from JSON
```

Comment:
You can probably use Jackson back reference annotation that is supported for this cases